### PR TITLE
feat: store metadata of newly created deployments

### LINKED ISSUE #24683: Store deployment record when creating new deployments
Currently we do nothing on the Deployment/Created event. To make deployments available long-term, we can start storing them again in the `DeploymentState`.

To avoid duplicating binary resources, we can just store the resource metadata.

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This creates a new applier for the Deployment/Created event that actually stores the record in the state. The records only contains the resource metadata.

Because we don't want to overwrite deployments, the processor now checks whether a deployment already exists before handling a distributed Deployment/Create command. If it does, we simply write a rejection.
This also means that we no longer re-create the resources needlessly, so that's a small performance win too.

:thought_balloon: I wasn't sure what to do about the old deployment handling where we remove the deployment from the state once fully distributed. This mechanism is no longer used since we switched to generalized command distribution but the code for it is still there, so there's a chance that we can still remove deployments. Could we remove the old processors/appliers now?

closes #24683

### CODE REVIEW

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Thanks @lenaschoenburg, very easy to review 👍 One thing needs to change

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ private void registerDeploymentAppliers(final MutableProcessingState state) {
 
     register(
         DeploymentIntent.CREATED, 1, new DeploymentCreatedV1Applier(state.getDeploymentState()));
-    register(DeploymentIntent.CREATED, 2, NOOP_EVENT_APPLIER);
+    register(DeploymentIntent.CREATED, 2, new DeploymentCreatedV2Applier());
```
[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java]: Maybe it's not? But I suggest just adding a V3. It's 2 minutes of work and guarantees it won't cause issues.

[the remark this answers is not in the document]
Comment (lenaschoenburg) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java]: Indeed, that was easy to change, see 989ac3b

**Review by remcowesterhoud — APPROVED** [review]
🎉

**Pull request merged**